waspc/src/Wasp/Node/Version.hs

```diff
@@ -39,79 +40,85 @@ data VersionCheckResult
-
--- | Gets the installed node version, if any is installed, and returns it.
---
--- Returns a string representing the error condition if node's version could
--- not be found.
-getUserNodeVersion :: IO (Either ErrorMessage SV.Version)
-getUserNodeVersion = do
-  -- Node result is one of:
-  -- 1. @Left processError@, when an error occurs trying to run the process
-  -- 2. @Right (ExitCode, stdout, stderr)@, when the node process runs and terminates
-  nodeResult <-
-    (Right <$> P.readProcessWithExitCode "node" ["--version"] "")
-      `catchIOError` ( \e ->
-                         if isDoesNotExistError e
-                           then return $ Left nodeNotFoundMessage
-                           else return $ Left $ makeNodeUnknownErrorMessage e
-                     )
-  return $ case nodeResult of
-    Left procErr ->
-      Left
-        ( unlines
-            [ "Running `node --version` failed.",
-              indent 2 procErr,
-              "Make sure you have `node` installed and in your PATH."
-            ]
-        )
-    Right (ExitFailure code, _, stderr) ->
-      Left
-        ( unlines
-            [ "Running `node --version` failed (exit code " ++ show code ++ "):",
-              indent 2 stderr
-            ]
-        )
-    Right (ExitSuccess, stdout, _) ->
-      parseNodeVersionOutput stdout
-        & left
-          ( \e ->
-              "Wasp failed to the parse `node` version provided by `node --version`.\n"
-                <> (show e <> "\n")
-                <> "This is most likely a bug in Wasp, please file an issue."
-          )
-
--- | Extracts node version from the output of `node --version`.
-parseNodeVersionOutput :: String -> Either P.ParseError SV.Version
-parseNodeVersionOutput = P.parse nodeVersionParser ""
+checkUserNodeAndNpmMeetWaspRequirements :: IO VersionCheckResult
+checkUserNodeAndNpmMeetWaspRequirements = do
+  nodeResult <- checkUserNodeVersion
+  npmResult <- checkUserNpmVersion
+  return $ case (nodeResult, npmResult) of
+    (VersionCheckSuccess, VersionCheckSuccess) -> VersionCheckSuccess
+    (VersionCheckFail nodeError, _) -> VersionCheckFail nodeError
+    (_, VersionCheckFail npmError) -> VersionCheckFail npmError
+
+checkUserNodeVersion :: IO VersionCheckResult
+checkUserNodeVersion = checkUserToolVersion "node" ["--version"] oldestWaspSupportedNodeVersion
+
+checkUserNpmVersion :: IO VersionCheckResult
+checkUserNpmVersion = checkUserToolVersion "npm" ["--version"] oldestWaspSupportedNpmVersion
+
+checkUserToolVersion :: String -> [String] -> SV.Version -> IO VersionCheckResult
+checkUserToolVersion commandName commandArgs oldestSupportedToolVersion = do
+  userVersion <- getToolVersionFromCommandOutput commandName commandArgs
+  return $ case userVersion of
+    Left errorMsg -> VersionCheckFail errorMsg
+    Right version
+      | version >= oldestSupportedToolVersion -> VersionCheckSuccess
+      | otherwise -> VersionCheckFail $ makeVersionMismatchErrorMessage version
```

Comment:
Naming suggestion:
  - `userVersion` -> `userVersionOrError`
  - `version` -> `userVersion`

Whenever two variables can switch names and still be correct, chances are we're abusing the English languge to avoid dealing with naming :)

More examples of this:
```ts
const directory = foo()
const dir = bar(directory)

// ...

const doesExist = exists()
```